airflow/contrib/operators/gcs_operator.py

```diff
@@ -0,0 +1,112 @@
+    so you can't create a bucket with a name that is already in use.
+
+        .. seealso::
+            For more information, see Bucket Naming Guidelines:
+            https://cloud.google.com/storage/docs/bucketnaming.html#requirements
+
+    :param bucket_name: The name of the bucket.
+    :type bucket_name: string
+    :param storage_class: This defines how objects in the bucket are stored
+            and determines the SLA and the cost of storage. Values include
+
+            - ``MULTI_REGIONAL``
+            - ``REGIONAL``
+            - ``STANDARD``
+            - ``NEARLINE``
+            - ``COLDLINE``.
+            If this value is not specified when the bucket is
+            created, it will default to STANDARD.
+    :type storage_class: string
+    :param location: The location of the bucket.
+        Object data for objects in the bucket resides in physical storage
+        within this region. Defaults to US.
+
+        .. seealso::
+            https://developers.google.com/storage/docs/bucket-locations
+
+    :type location: string
+    :param project_id: The ID of the GCP Project.
+    :type project_id: string
+    :param labels: User-provided labels, in key/value pairs.
+    :type labels: dict
+    :param google_cloud_storage_conn_id: The connection ID to use when
+        connecting to Google cloud storage.
+    :type google_cloud_storage_conn_id: string
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must
+        have domain-wide delegation enabled.
+    :type delegate_to: string
+
+    **Example**:
+        The following Operator would create a new bucket ``test-bucket``
+        with ``MULTI_REGIONAL`` storage class in ``EU`` region ::
+
+            CreateBucket = GoogleCloudStorageCreateBucket(
+                task_id='CreateNewBucket',
+                bucket_name='test-bucket',
+                storage_class='MULTI_REGIONAL',
+                location='EU',
+                labels={'env': 'dev', 'team': 'airflow'},
+                google_cloud_storage_conn_id='airflow-service-account'
+            )
+    """
+
+    template_fields = ('bucket_name', 'storage_class',
+                       'location', 'project_id')
+    ui_color = '#f0eee4'
+
+    @apply_defaults
+    def __init__(self,
+                 bucket_name,
+                 storage_class='MULTI_REGIONAL',
+                 location='US',
+                 project_id=None,
+                 labels=None,
+                 google_cloud_storage_conn_id='google_cloud_storage_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+        super(GoogleCloudStorageCreateBucket, self).__init__(*args, **kwargs)
+        self.bucket_name = bucket_name
+        self.storage_class = storage_class
+        self.location = location
+        self.project_id = project_id
+        self.labels = labels
```

Comment:
mind adding a default airflow system labe? e.g., https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/operators/dataflow_operator.py#L116

Reply:
@fenglu-g Can do that. But do you want me to replace `labels=None` to `labels={'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')})` or update whatever labels are passed with `labels.update({'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')}))`

My only concern is some companies have a strict policy for using labels to track their pricing and already have a naming convention set for their labels. 

What's your suggestion? Should we be passing the `airflow-version` label if no labels are passed or just add this key to whatever labels they pass?

Reply:
Or do you want me to add something like:
```python
        if self.labels is not None:
            self.labels.update(
                {'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')}
            )
```

Reply:
The latter, a combination of both airflow system labels and customer supplied labels. The idea is to measure the impact of Airflow.

Reply:
Done